src/lib.rs

```diff
@@ -565,8 +614,71 @@ pub fn get_lines_from_osm(pbf: &mut OsmPbfReader) -> Vec<Line> {
         .collect()
 }
 
+pub fn get_routes_from_stop(routes: &Vec<Route>, stop_point: &StopPoint) -> Vec<Route> {
```

Comment:
Here too, you don't need to reallocate a new `Route`.
```suggestion
pub fn get_routes_from_stop<'a>(routes: &'a Vec<Route>, stop_point: &StopPoint) -> Vec<&'a Route> {
```

This involves a little bit more changes to the code. See https://github.com/prhod/osm-transit-extractor/pull/1.